scripts/stats_std.py

```diff
@@ -0,0 +1,80 @@
+import pysam
+import pysamstats
+import numpy as np
+import subprocess
+
+def threshold(reads):
+	mean = np.mean(reads)
+	#print(mean)
+	std = np.std(reads)
+	#print(std)
+	thresh_pos = mean + (3 * std)
+	thresh_neg = mean - (3 * std)
+	return thresh_pos, thresh_neg
+
+def mean_of_array(reads,start, pos):
+	mean = np.mean(reads[start:pos], dtype=int)
+	return mean
+
+def main():
+	mybam = pysam.AlignmentFile('UMB.aligned.sorted.bam')
+
+	# iterate over statistics, one record at a time
+	#stats = pysamstats.load_coverage(mybam, chrom='NZ_CP023386.1', start= 2000000, end=3000000)
+	stats = pysamstats.load_coverage(mybam)
+
+	reads = stats.reads_all
+	position = stats.pos
+	chrom = stats.chrom
+
+	print(len(reads))
+	print(len(position))
+	print(len(chrom))
+
+	thresh_up, thresh_down = threshold(reads)
+	#print(thresh_up)
+	#print(thresh_down)
+
+	start = 0
+	big_start = 0
+	last_end = 0
+
+	file = 'output_std.txt'
+
+	with open(file, "w") as f:
+		print("Index Chrom Pos Thresh Diff", file=f)
+		for i in range(len(reads)):
```

Comment:
Python's range function can also accept a `step` parameter, which allows you to quickly iterate over blocks: https://docs.python.org/3/library/functions.html#func-range

E.g. `range(0, CHROMOSOME_LENGTH, 1000)` to iterate over 1000 bp blocks.

That will make the code below a bit simpler (e.g. the `i % 1000` is not necessary anymore).